src/deps/uws.zig

```diff
@@ -1744,36 +1651,22 @@ pub fn NewSocketHandler(comptime is_ssl: bool) type {
             return this.socket.isClosed(comptime is_ssl);
         }
 
-        pub fn close(this: ThisSocket, code: CloseCode) void {
+        pub fn close(this: ThisSocket, code: Socket.CloseCode) void {
             return this.socket.close(comptime is_ssl, code);
         }
         pub fn localPort(this: ThisSocket) i32 {
-            switch (this.socket) {
-                .connected => |socket| {
-                    debug("us_socket_local_port({d})", .{@intFromPtr(socket)});
-                    return us_socket_local_port(
-                        comptime ssl_int,
-                        socket,
-                    );
-                },
-                .pipe, .upgradedDuplex, .connecting, .detached => return 0,
-            }
+            return switch (this.socket) {
+                .connected => |socket| socket.localPort(is_ssl),
+                .pipe, .upgradedDuplex, .connecting, .detached => 0,
+            };
         }
-        pub fn remoteAddress(this: ThisSocket, buf: [*]u8, length: *i32) void {
-            switch (this.socket) {
-                .connected => |socket| {
-                    debug("us_socket_remote_address({d})", .{@intFromPtr(socket)});
-                    return us_socket_remote_address(
-                        comptime ssl_int,
-                        socket,
-                        buf,
-                        length,
-                    );
-                },
-                .pipe, .upgradedDuplex, .connecting, .detached => return {
-                    length.* = 0;
-                },
-            }
+
+        /// `buf` cannot be longer than 2^31 bytes long.
+        pub fn remoteAddress(this: ThisSocket, buf: []u8) ?[]const u8 {
+            return switch (this.socket) {
+                .connected => |sock| sock.remoteAddress(is_ssl, buf) catch unreachable, // fixme
```

Comment:
do not ever use `unreachable` on code that is reachable. this is what `@panic` is for.